packages/next/server/image-optimizer.ts

```diff
@@ -420,12 +420,14 @@ function setResponseHeaders(
   isDev: boolean
 ) {
   res.setHeader('Vary', 'Accept')
-  res.setHeader(
-    'Cache-Control',
-    isStatic
-      ? 'public, max-age=315360000, immutable'
-      : `public, max-age=${isDev ? 0 : maxAge}, must-revalidate`
-  )
+  if (!res.getHeader('Cache-Control')) {
+    res.setHeader(
+      'Cache-Control',
+      isStatic
```

Comment:
I don't think we should allow overriding the static header. This can lead to accidentally de-optimization.

I would prefer PR #26739 and PR #23328 to handle this use case and avoid possible footguns.

Reply:
Hi @styfle , I agree that the solution you propose is neater. I'll close this PR in favour of the ones you mentioned. Thanks for your feedback and glad to see this issue is finally solved.